psi4/src/psi4/libfock/PKmanagers.cc

```diff
@@ -107,24 +110,48 @@ std::shared_ptr<PKManager> PKManager::build_PKManager(std::shared_ptr<PSIO> psio
         ncorebuf = 3;
     }
 
+    // determine which sub-algorithm to use
     bool do_reord = false;
     bool do_yosh = false;
     bool do_incore = false;
-    if (options["PK_ALGO"].has_changed()) {
-        if (algo == "REORDER") {
-            do_reord = true;
-        } else if (algo == "YOSHIMINE") {
-            do_yosh = true;
-        }
-    } else {
+
+    // specify particular out-of-core subalgorithm...
+    if (subalgo == "OOC_REORDER") {
+	do_reord = true;
+    } else if (subalgo == "OOC_YOSHIMINE") {
+        do_yosh = true;
+
+    // ...or automatically select an out-of-core subalgorithm...
+    } else if (subalgo == "OUT_OF_CORE") {
         if (algo_factor * memory > pk_size) {
             do_reord = true;
         } else {
             do_yosh = true;
         }
-    }
 
-    if (ncorebuf * pk_size < memory && !noincore) do_incore = true;
+    // ...or force the in-core algorithm...
+    } else if (subalgo == "INCORE") {
+	// throw an exception if in-core is forced, but not enough memory is allocated
+        if (ncorebuf * pk_size > memory) {
+            throw PSIEXCEPTION("SCF_SUBTYPE=INCORE was specified, but there is not enough memory to do in-core! Increase the amount of memory allocated to Psi4 or allow for out-of-core to be used.\n");
+        } else {
+            do_incore = true;
+	}
+
+    // ...or just let psi4 pick any subalgorithm
+    } else if (subalgo == "AUTO") {
+        if (ncorebuf * pk_size < memory) {
+            do_incore = true;
+	} else if (algo_factor * memory > pk_size) {
```

Comment:
logic lgtm, but the formatting/spacing is deceptive. :-)

Reply:
`do_record` on line 120 should be a tab inwards. `} else if` on 146 should be aligned with 144. 148 and 149 are two tabs apart. It's very common in your PRs, so you might want to look at your editor's whitespace settings.

Reply:
Ah, I see. Yeah, that's something I've been meaning to do, and just haven't gotten around to. I'll do that now and recommit with proper spacing.

Reply:
All right, it should be better now, I think